common/models/application.js

```diff
@@ -81,8 +81,14 @@ module.exports = function(Application) {
    * A hook to generate keys before creation
    * @param next
    */
-  Application.beforeCreate = function(next) {
-    var app = this;
+  Application.observe('before save', function(ctx, next) {
+    if (!ctx.instance) {
+      // Example: Application.updateAll() and Application.updateOrCreate()
+      var msg = 'Application object does not support data-based updates';
+      return next(new Error(msg));
```

Comment:
Can you please post a short snippet of code that you need to keep working with my change in place?